pyTigerGraph/pytgasync/pyTigerGraphAuth.py

```diff
@@ -0,0 +1,201 @@
+
+            Args:
+                alias:
+                    One or more alias(es) of secret(s).
+                ignoreErrors:
+                    Ignore errors arising from trying to drop non-existent secrets.
+
+            Raises:
+                `TigerGraphException` if a non-existent secret is attempted to be dropped (unless
+                `ignoreErrors` is `True`). Re-raises other exceptions.
+        """
+        logger.info("entry: dropSecret")
+        if logger.level == logging.DEBUG:
+            logger.debug("params: " + self._locals(locals()))
+
+        if isinstance(alias, str):
+            alias = [alias]
+        cmd = """
+            USE GRAPH {}""".format(self.graphname)
+        for a in alias:
+            cmd += """
+                DROP SECRET {}""".format(a)
+        res = await self.gsql(cmd)
+        if "Failed to drop secrets" in res and not ignoreErrors:
+            raise TigerGraphException(res)
+
+        if logger.level == logging.DEBUG:
+            logger.debug("return: " + str(res))
+        logger.info("exit: dropSecret")
+
+        return res
+
+    async def _newToken(self, secret: str = None, lifetime: int = None, token=None, _method=None) -> Union[tuple, str]:
+        method, url, alt_url, authMode, data, alt_data = self._prep_newToken(
+            secret=secret, lifetime=lifetime, token=token)
+        #  _method Used for delete and refresh token
+
+        # method == GET when using old version since _prep_newToken() gets the method for getting a new token for a version
+        if method == "GET":
+            if _method:
+                method = _method
+
+            # Use TG < 3.5 format (no json data)
+            res = await self._req(method, url, authMode=authMode, data=data, resKey=None)
+            mainVer = 3
+        else:
+            if _method:
+                method = _method
+
+            # Try using TG 4.1 endpoint first, if url not found then try <4.1 endpoint
+            try:
+                res = await self._req(method, url, authMode=authMode, data=data, resKey=None, jsonData=True)
+                mainVer = 4
+            except httpx.HTTPError as e:
+                if e.response.status_code == 404 or e.response.status_code == 405:
+                    res = await self._req(method, alt_url, authMode=authMode, data=alt_data, resKey=None)
+                    # res['token'] = res['results']['token'] # bringing token out to outer dict level for consistency, can't just use resKey="results" since want to preserve error message
+                    mainVer = 3
+                else:
+                    raise e
+
+        # uses mainVer instead of _versionGreaterThan4_0 since you need a token for verson checking
+        return res, mainVer
+
+    async def newGetToken(self, secret: str = None, setToken: bool = True, lifetime: int = None) -> Union[tuple, str]:
```

Comment:
Should this be prefaced with new, or is it just getToken. Same question for refresh and delete

Reply:
I reviewed all the TODO comments in the `pytgasync` folder. It appears they were copied directly from the synchronous functions and are outdated, as they haven't been implemented in either the async or sync versions. 

Regarding functions prefixed with "new," it seems Dyllan has broken down the `getToken` function into 'newGetToken' with smaller helper functions: `_prep_newToken`, `_newToken`, and `_parse_newToken`, which can be reused in "newDeleteToken", "newRefreshToken". These new functions are used across unit tests. 

Let me try to comment out "getToken" function, and rename "newGetToken" to "getToken", and run another round test. 

I agree that we might have to reorganize and clean up pyTG as a whole. We can discuss it with the team and add it to sprint planning.

Reply:
So the api is consistent across sync and async, they should not have the "new" prefix

Reply:
cleaned up the code by removing new prefix and duplicates, update the functions across different functions.